lib/pure/collections/deques.nim

```diff
@@ -20,6 +20,11 @@
 ## access, unless your program logic guarantees it indirectly.
 ##
 ## .. code-block:: Nim
+##   import random
```

Comment:
I really dislike these long code blocks which are meant to be examples but they make it really difficult for readers to find what they are looking for. It surely is better than nothing, but please consider separating this large sample of code into headings similar to how it's done in the [httpclient module](http://nim-lang.org/docs/httpclient.html).